classes/Link.php

```diff
@@ -772,7 +773,86 @@ public function getAdminLink($controller, $withToken = true, $sfRouteParams = ar
 
         $idLang = Context::getContext()->language->id;
 
-        return $this->getBaseLink().basename(_PS_ADMIN_DIR_).'/'.Dispatcher::getInstance()->createUrl($controller, $idLang, $params);
+        return $this->getAdminBaseLink().basename(_PS_ADMIN_DIR_).'/'.Dispatcher::getInstance()->createUrl($controller, $idLang, $params);
+    }
+
+    /**
+     * @param int|null  $idShop
+     * @param bool|null $ssl
+     * @param bool      $relativeProtocol
+     * @return string
+     * @throws PrestaShopDatabaseException
+     */
+    public function getAdminBaseLink($idShop = null, $ssl = null, $relativeProtocol = false)
+    {
+        static $force_ssl = null;
+
+        if ($ssl === null) {
```

Comment:
Try to be consistent with the order of your comparisons.

Reply:
yes my bad, this was former code I did not update